docs/decisions/000x-internal-editor-testability-decisions.md

```diff
@@ -0,0 +1,136 @@
+# Internal editor testability decision
```

Comment:
nit: as this ADR contains more information about general code structure in the name of improved testability, it might be worth noting that in the title of the ADR, so those confused by the code structure in this repo know to look here.

Reply:
"Testability decisions" to the lay public might get skipped over because to them it means "we are shooting for 100% and we use Jest"

Reply:
"decision on structuring code for testability" maybe or something similar